src/Foundation/NSAction.cs

```diff
@@ -55,27 +56,26 @@ public void Apply ()
 	}
 
 	// Use this for synchronous operations
-	[Register ("__MonoMac_ActionDispatcher")]
-	internal sealed class ActionDispatcher : NSObject {
+	[Register ("__MonoMac_NSActionDispatcher2")]
+	internal sealed class NSActionDispatcher2 : NSObject
+	{
 		public const string SelectorName = "xamarinApplySelector";
 		public static readonly Selector Selector = new Selector (SelectorName);
 
-		readonly Action action;
+		readonly SendOrPostCallback d;
+		readonly object state;
 
-		public ActionDispatcher (Action action)
+		public NSActionDispatcher2 (SendOrPostCallback d, object state)
 		{
-			if (action == null)
-				throw new ArgumentNullException ("action");
-
-			this.action = action;
-			IsDirectBinding = false;
```

Comment:
you can't remove `IsDirectBinding = false;` since there's managed state in the instance

Reply:
Ah, my bad.